### DISCUSSION TITLE
Use local container registry (e.g. Harbor) instead of Docker Hub

### DISCUSSION THREAD

**Discussion by szpak** [renovatebot/renovate#23106] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.40.0

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Wanted end result.

In some projects in `docker-compose.yml` files people use old-fashioned reference to docker-hub images:

```
version: '3.9'
services:
  db:
    image: postgres:14.6-alpine
    container_name: db
  wiremock:
    image: wiremock/wiremock:2.33.2
    container_name: wiremock
...
```

It (usually) works locally, but on CI when Renovate is executed it generates "Too Many Requests" problem with rate-limiting in Docker Hub. The best would to add the local Docker registry proxy (here Harbor) in every problematic `docker-compose.yml`, but Renvoate is very flexible and it would be possible to achieve in the global configuration for all the projects?

The expected end result is to look for:
- `harbor.example.com/proxy/wiremock/wiremock` instead of `wiremock/wiremock` in DockerHub
- `harbor.example.com/proxy/library/postgres` instead of just `postgres` (the official package) in DockerHub (mind the "library" group in Harbor)

Renovate could propose a change to harbor, or do it transparantly to keep the original construction (and just bump the version).

Can it be achieved just for docker-compose (or if not, for docker datasource in general)?

### What you tried so far.

I tried to change the oroginal definitions in `docker-compose.yml` manually, but it would take a while...

**Comment by viceice** [comment]
use the renovate replacement feature to automatically update those images to the proxy.

**Reply by rarkins** [reply]
I thought plain old registryUrls work with Docker Hub images

**Reply by viceice** [reply]
maybe 🤷‍♂️ never tested
